### PR TITLE
GH-71 render bind clause with SPARQLQueryRenderer

### PR DESCRIPTION
**Pull request by jeenbroekstra**
GitHub issue resolved: #71  

Briefly describe the changes proposed in this PR:

----
PR Author Checklist (see the [contributor guidelines](https://github.com/eclipse/rdf4j/blob/main/.github/CONTRIBUTING.md) for more details):

 - [ ] my pull request is [self-contained](https://rdf4j.org/documentation/developer/merge-strategy/#self-contained-changes-pull-requests-and-commits)
 - [ ] I've added tests for the changes I made
 - [ ] I've applied [code formatting](https://github.com/eclipse/rdf4j/blob/main/.github/CONTRIBUTING.md#code-formatting) (you can use `mvn process-resources` to format from the command line)
 - [ ] I've [squashed](https://rdf4j.org/documentation/developer/squashing) my commits where necessary 
 - [ ] every commit message starts with the issue number (GH-xxxx) followed by a meaningful description of the change

### CODE REVIEW

File: core/sail/solr/.factorypath [hunk 1]
```diff
@@ -0,0 +1,109 @@
+<factorypath>
+    <factorypathentry kind="VARJAR" id="M2_REPO/org/apache/commons/commons-text/1.9/commons-text-1.9.jar" enabled="true" runInBatchMode="false"/>
```
[the remark this answers is not in the document]
Comment (JervenBolleman) [reply core/sail/solr/.factorypath:2]: It is in the last commit by @RedCrusaderJr.  I think the 
```
git rm core/sail/solr/.factorypath 
git rm testsuites/benchmark/.factorypath
git commit -m "GH-71 Add implementation for rendering BIND construct"
git push -f
```
will fix this.  #GH-2989 should avoid this in the future.

**Comment by JervenBolleman** [comment]
@RedCrusaderJr very good job, big improvement on what I did already!

**Comment by jeenbroekstra** [comment]
There is a currently failing test:

```
org.opentest4j.AssertionFailedError: 
expected: <select ?b
where {
  bind(1 as ?b).
}> but was: <select ?b
where {
  bind(?b as ?b).
  bind(1 as ?b).
}>
	at org.eclipse.rdf4j.queryrender.SPARQLQueryRenderTest.testBind(SPARQLQueryRenderTest.java:18)
```

As also discussed on gitter though, this is caused by the fact that the SPARQL parser introduces a redundant `ExtensionElem` in the algebra (see #2985). As soon as the fix for that is approved and merged we should rebase this.

File: core/sail/solr/.factorypath (hunk 1 above)
[the remark this answers is not in the document]
Comment (RedCrusaderJr) [reply core/sail/solr/.factorypath:2]: I am a clumsy git user... Sorry for the inconvenience :(

**Comment by jeenbroekstra** [comment]
@RedCrusaderJr the fix for GH-2985 (the redundant ExtensionElems) has been merged into the main branch. It would be useful if we could rebase your branch so that it includes those changes. That way we can makes sure nothing breaks. In the same process you could "squash" your commits a little, to clean up the history a little bit, and making sure that your commit messages are all good (starting with issue number and so on).

Are you aware how you can rebase and squash your branch? It's a little bit tricky if you haven't done it before, so I recommend that before you start you run `git checkout -b GH-71-BIND-backup` (from your own branch) so you have a backup if something goes wrong (and don't forget to switch back to your original branch before starting the rebase). For tips on how to work with interactive rebase/squashing, see [this page](https://rdf4j.org/documentation/developer/squashing/). 

If you'd rather not do this, just let me know, I'm also happy to take care of it for you.

**Comment by jeenbroekstra** [comment]
@JervenBolleman this looks good to me - can you take another look and say if you're happy for this to be merged? Obviously there's more things we can fix so I won't close GH-71, but this is a good start, and self-contained.

**Comment by JervenBolleman** [comment]
@RedCrusaderJr thank you for picking this up. @jeenbroekstra I think it is great LGTM!

**Comment by jeenbroekstra** [comment]
Actually @RedCrusaderJr adviced me on gitter that they're still fixing a few things.

**Comment by jeenbroekstra** [comment]
There's a couple of failing tests, likely caused by minor reordering of elements in the algebra tree (breaking tests that do string comparisons). I'll look at them in turn and fix where appropriate.

**Pull request merged**